src/realm/query_expression.cpp

```diff
@@ -519,11 +519,16 @@ void LinkCount::evaluate(size_t index, ValueBase& destination)
                 const Obj obj = m_link_map.get_target_table()->get_object(links[i]);
                 auto val = obj._get<int64_t>(m_column_key.get_index());
                 size_t s;
-                if (val & 1) {
+                if (m_column_key.get_type() == col_type_Link && !m_column_key.is_collection()) {
```

Comment:
Do I read this correctly? You are trying to avoid to deal with collections of links? Right?

Reply:
Yes. We must handle single link columns separately. That was the problem.